api/src/test/java/com/epam/pipeline/manager/pipeline/RunScheduleManagerTest.java

```diff
@@ -103,179 +121,242 @@ public void setUp() {
+        verify(mockScheduleProviderManager, times(2)).getProvider(ScheduleType.RUN_CONFIGURATION);
     }
 
+
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
     public void testCreateRunScheduleWithExistentCronExpression() {
+        when(mockScheduleProviderManager.getProvider(ScheduleType.PIPELINE_RUN)).thenReturn(mockPipelineRunScheduleProvider);
+        when(mockAuthManager.getCurrentUser()).thenReturn(mockPipelineUser);
+
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
                 Collections.singletonList(testRunScheduleVO));
         testRunScheduleVO.setAction(RunScheduledAction.RESUME);
+
+        doReturn(Collections.singletonList(createRunSchedule())).
+                when(mockRunScheduleDao).loadAllRunSchedulesBySchedulableIdAndType(RUN_ID, ScheduleType.PIPELINE_RUN);
+
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
                 Collections.singletonList(testRunScheduleVO));
     }
 
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
-    public void  createSchedulesWithIdenticalExpressions() {
+    public void createSchedulesWithIdenticalExpressions() {
         final PipelineRunScheduleVO runScheduleVO = getRunScheduleVO(RunScheduledAction.PAUSE, CRON_EXPRESSION1);
         final PipelineRunScheduleVO runScheduleVO2 = getRunScheduleVO(RunScheduledAction.RESUME, CRON_EXPRESSION1);
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
                 Arrays.asList(runScheduleVO, runScheduleVO2));
     }
 
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
-    public void  createSchedulesWithWrongAction() {
+    public void createSchedulesWithWrongAction() {
+        doThrow(IllegalArgumentException.class).when(mockScheduleProviderManager).getProvider(ScheduleType.RUN_CONFIGURATION);
         final PipelineRunScheduleVO runScheduleVO = getRunScheduleVO(RunScheduledAction.RESUME, CRON_EXPRESSION1);
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.RUN_CONFIGURATION,
                 Collections.singletonList(runScheduleVO));
     }
 
     @Test
-    @WithMockUser(username = USER_OWNER)
     public void testUpdateRunSchedules() {
+        when(mockScheduleProviderManager.getProvider(ScheduleType.PIPELINE_RUN)).thenReturn(mockPipelineRunScheduleProvider);
+        when(mockAuthManager.getCurrentUser()).thenReturn(mockPipelineUser);
+
         final List<RunSchedule> schedules =
-            runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                    Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
+                runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
+                        Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
+
+        schedules.get(0).setId(1L);
+        schedules.get(1).setId(2L);
```

Comment:
Again, as long as we write a unit test  then there is no actual need to create schedules first using `runScheduleManager`. Just mock everything you need and then test only a single method - `updateSchedules`.

Same for all update and delete tests in the class.